### DISCUSSION TITLE
new version generate style filename like app.css.2c9b54c9.css not app.2c9b54c9.css

### DISCUSSION THREAD

**Discussion by nyrf** [ElMassimo/vite_ruby#66] [Q&A] [answered]
I found after update to the newest version, the style file generated like `app.css.2c9b54c9.css`, is there a way to make the filename as`app.2c9b54c9.css` like before ? Thanks.

**Comment by ElMassimo** [comment]
Hi, would you provide more information about your use case?

**Reply by nyrf** [reply]
> Hi, would you provide more information about your use case?

Sorry for my weak english, i mean after build, `app/frontend/entrypoints/app.css` will be compiled to `public/vite/assets/app.css.2c9b54c9.css`, i remembered that file should be `public/vite/assets/app.2c9b54c9.css`.

**Reply by ElMassimo** [reply]
That is intentional in order to support scss entrypoints. It should be possible to avoid the extra extension for `.css`, but it's necessary at the moment for other extensions.

**Reply by nyrf** [reply]
Thanks, I've used @doabit codes, it works.

**Comment by doabit** [comment, marked as answer]
Try add this codes to `vite.config.js`, i think it works

`vite.config.js`

```js
  build: {
    rollupOptions: {
      output: {
        assetFileNames: (assetInfo) => {
          let name = assetInfo.name
          const reg = /(\.\w+)\.css$/
          const result = reg.exec(name)
          if (result) {
            name = name.replace(result[0], '')
          } else {
            name = name.substr(0, name.lastIndexOf('.'))
          }
          return `assets/${name}.[hash][extname]`
        }
      }
    }
  }
```

**Reply by nyrf** [reply]
Works, thanks.

**Reply by ElMassimo** [reply]
@doabit @nyrf This has been addressed in 3.0.

https://github.com/ElMassimo/vite_ruby/pull/116

You would no longer need to provide `assetFileNames` to remove the prefix.

**Reply by doabit** [reply]
@ElMassimo Great.
